src/components/Modal/index.jsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import './Modal.scss';
+import Signin from 'components/Signin';
+
+export class Modal extends React.Component {
+  /**
+    * @name Modal propTypes
+    * @type {propTypes}
+    *
+    * @param {Object} props - React PropTypes
+    *
+    * @property {Boolean} formTypes - toggles (false)signup or (true)signin form
+    * @property {Boolean} show - determines if modal should show or not
+    *
+    */
+  static propTypes = {
+    formType: PropTypes.bool.isRequired,
+    show: PropTypes.bool.isRequired
+  };
+
+  constructor(props) {
+    super(props);
+    const { formType, show } = this.props;
+    this.state = {
+      signin: formType,
+      show
+    };
+  }
+
+  /**
+    * Handles change form display action
+    * @method
+    *
+    * @return {void}
+    */
+  handleFormChange = () => {
+    const { signin } = this.state;
+    this.setState({ signin: !signin });
+  }
+
+  /**
+    * Handles close modal action
+    * @method
+    *
+    * @return {void}
+    */
+  handleClose = () => {
+    this.setState({ show: false });
+  }
+
+  render() {
+    const { signin, show } = this.state;
+    return (
+      <div className={(show) ? 'modal modal--show' : 'modal modal--hide'}>
+        <div className="modal__modal-content">
+          <button className="modal__close" type="button" onClick={this.handleClose}>&#x2715;</button>
+          {
+            signin ? (
+              <div className="modal__left">
+                <h2 className="modal__head">Share your stories</h2>
+                <h1 className="modal__welcome">Welcome back</h1>
+                <p className="modal__text">Sign into your account to serve you more personalised stories from creatives</p>
+              </div>
+            ) : (
+              <div className="modal__left">
+                <h2 className="modal__head">Share your stories</h2>
+                <h1 className="modal__welcome">Create your free account</h1>
+                <p className="modal__text">Sign into your account to serve you more personalised stories from creatives</p>
+              </div>
+            )
+          }
+          <div className="modal__right">
+            <h4 className="modal__heading">Select one to get started</h4>
+            <div className="modal__social-login">
+                IBUKUN COMPONENT
+            </div>
+            <div className="modal__seperator">
+              <p>or</p>
+            </div>
+            <div className="modal__authComponent">
```

Comment:
```suggestion
            <div className="modal__auth-component">
```
We will not have camel-cased class names. This is why BEM uses `__ `and `--`, so that you can have `_` and `-` for classnames. Kindly correct